### ISSUE TITLE
json_encode() of PublicKeyCredentialRequestOptions fails

### ISSUE THREAD

**Issue by ptmkenny** [web-auth/webauthn-framework#655]
### Version(s) affected

5.0

### Description

Works in 4.9 but not 5.0: `json_encode(\Webauthn\PublicKeyCredentialRequestOptions::create(random_bytes(32), allowCredentials: []))`

I need to json_encode() it to submit the request to my site via a [JSON-RPC](https://www.jsonrpc.org/specification) endpoint.

The [docs](https://webauthn-doc.spomky-labs.com/pure-php/authenticate-your-users) still say:

> The PublicKeyCredentialRequestOptions object is designed to be easily serialized into a JSON object. This will ease the integration into an HTML page or through an API endpoint.

### How to reproduce

With 4.9, I was using this code:

```
      $public_key_credential_request_options = PublicKeyCredentialRequestOptions::create(
        $this->webauthn->createChallenge(),
        allowCredentials: $allowed_credentials,
      );
      $this->tempstore->set(self::TEMP_STORE_REQUEST_OPTIONS, $public_key_credential_request_options);

      $stringified_json = json_encode($public_key_credential_request_options);
      if (is_string($stringified_json)) {
      }
```

After upgrading to 5.0, this fails because the value of json_encode() is false.

**Comment by albanx** [comment]
I had the same issue today, the only way to make it work is to serialize it and send as JSON string:

This is the example in the docs:

```
use Symfony\Component\Serializer\Encoder\JsonEncode;
use Symfony\Component\Serializer\Normalizer\AbstractObjectNormalizer;

// The serializer is the same as the one created in the previous pages
$jsonObject = $serializer->serialize(
    $publicKeyCredentialCreationOptions,
    'json',
    [ // Optional
        AbstractObjectNormalizer::SKIP_NULL_VALUES => true,
        JsonEncode::OPTIONS => JSON_THROW_ON_ERROR,
    ]
);
```

**Comment by Spomky** [comment]
> After upgrading to 5.0, this fails because the value of json_encode() is false.

Indeed, `json_encode` should not be used as in 5.0.+ the objects do not implement the `JsonSerializable` interface.
Please use [the Serializer](https://webauthn-doc.spomky-labs.com/pure-php/input-loading#the-serializer) instead.

**Comment by ptmkenny** [comment]
Thank you for the updates.  Also, `Webauthn\PublicKeyCredentialSource::jsonSerialize()` was removed, but this isn't listed in the [migration docs](https://webauthn-doc.spomky-labs.com/v4.9/migration/from-v3.x-to-v4.0-1).
